custom_components/wattbox/button.py

```diff
@@ -0,0 +1,117 @@
+"""Button platform for wattbox."""
+
+import logging
+import re
+from typing import Any, List
+
+from homeassistant.components.button import ButtonDeviceClass, ButtonEntity
+from homeassistant.const import CONF_NAME, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from pywattbox.base import BaseWattBox, Outlet
+
+from .const import (
+    CONF_NAME_REGEXP,
+    CONF_SKIP_REGEXP,
+    DOMAIN_DATA,
+    RESTART_ICON,
+)
+from .entity import WattBoxEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_regex(config: ConfigType, key: str) -> re.Pattern[str] | None:
+    regexp_str: str = config.get(key, "")
+    if regexp_str:
+        try:
+            return re.compile(regexp_str)
+        except re.error:
+            _LOGGER.error("Invalid %s: %s", key, regexp_str)
+    return None
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType,
+) -> None:
+    """Setup button platform."""
+    name: str = discovery_info[CONF_NAME]
+
+    entities: List[WattBoxEntity] = []
+    wattbox: BaseWattBox = hass.data[DOMAIN_DATA][name]
+
+    name_regexp = validate_regex(config, CONF_NAME_REGEXP)
+    skip_regexp = validate_regex(config, CONF_SKIP_REGEXP)
+
+    for i, outlet in wattbox.outlets.items():
+        outlet_name = outlet.name or ""
+
+        # Skip outlets if they match regex
+        if skip_regexp and skip_regexp.search(outlet_name):
+            _LOGGER.debug("Skipping outlet #%s - %s", i, outlet_name)
+            continue
+
+        if name_regexp:
+            if matched := name_regexp.search(outlet_name):
+                outlet_name = matched.group()
+                try:
+                    outlet_name = matched.group(1)
+                except re.error:
+                    pass
+
+        _LOGGER.debug("Adding outlet reset #%s - %s", i, outlet_name)
+        entities.append(WattBoxResetButton(hass, name, i, outlet_name))
```

Comment:
Could probably do something shared with this, since switch and button have the same logic except the class they implement. That would ensure the name / skip logic doesn't get out of sync between them.